crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals_elif_else.md

```diff
@@ -0,0 +1,42 @@
+# Narrowing for conditionals with elif and else
+
+## Positive contributions become negative in elif-else blocks
+
+```py
+def int_instance() -> int: ...
+
+x = int_instance()
+
+if x == 1:
+    reveal_type(x)  # revealed: Literal[1]
```

Comment:
We've previously discussed that we can't actually safely do this narrowing, because `x` here could be an instance of some subclass of `int` that overrides `__eq__` to compare equal to `1`.

Perhaps experience will tell us that we are being too conservative here, and we should just ignore that possibility; but I'd rather not do that until/unless it's clear that we need to.